### PR TITLE
fix: Adds CLI help flag. Fixes #31

### LINKED ISSUE #31: Add support for CLI help (-h|--help)
I expect any CLI to have these kinds of options as a quick way to check usage, such as if there's any kind of debug flag.

### PR DESCRIPTION
**Pull request by grant**
Adds a CLI help flag (-h, --help). Fixes #31.

## Example

```sh
granttimmerman @ ~/Documents/trash/testff
❤ functions-framework --help
Usage:
  functions-framework --target=helloWorld --port=8080
Documentation:
  https://github.com/GoogleCloudPlatform/functions-framework-nodejs
```

CC: @grayside

### CODE REVIEW

**Comment by grant** [comment]
Hey @grayside, can you review this PR?

File: src/index.ts [hunk 1]
```diff
@@ -73,6 +73,17 @@ if (SIGNATURE_TYPE === undefined) {
...
 
+// CLI Help Flag
+if (process.argv[2] === '-h' || process.argv[2] === '--help') {
+  console.error(
+    `Example usage:
+  functions-framework --target=helloWorld --port=8080
+Documentation:
+  https://github.com/GoogleCloudPlatform/functions-framework-nodejs`
+  );
+  process.exit(1);
```
Comment (grayside) [inline src/index.ts]: I would expect a help command to exit on success,  errors reserved for outputting on another invalid command.

Comment (grant) [reply src/index.ts]: Fixed.

**Comment by grant** [comment]
Yes, could be improved besides looking for `process.argv[2]`.
If we want, we can create a more detailed GitHub issue that can be the basis for all function frameworks; but we have to be careful of implementation across all of them.

Merging.

**Pull request merged**